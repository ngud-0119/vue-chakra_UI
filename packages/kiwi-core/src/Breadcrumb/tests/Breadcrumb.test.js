import { Breadcrumb, BreadcrumbItem, BreadcrumbLink, BreadcrumbSeparator } from '../'
import { render } from '@/tests/test-utils'
import CompositionApi from '@vue/composition-api'

const renderComponent = (props) => {
  const base = {
    components: {
      Breadcrumb, BreadcrumbItem, BreadcrumbLink, BreadcrumbSeparator
    },
    template: `
    <div>
      <Breadcrumb :add-separator="false">
        <BreadcrumbItem>
          <BreadcrumbLink href="#">Breadcrumb 1</BreadcrumbLink>
          <BreadcrumbSeparator color="tomato" font-size="10px" font-weight="bold" />
        </BreadcrumbItem>
        <BreadcrumbItem>
          <BreadcrumbLink href="#">Breadcrumb 2</BreadcrumbLink>
          <BreadcrumbSeparator color="firebrick" font-size="20px" font-weight="bold" />
        </BreadcrumbItem>
      </Breadcrumb>
    </div>
   `,
    ...props
  }
  return render(base, {}, vue => {
    vue.use(CompositionApi)
  })
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})

it('should display custom seperator ', () => {
  const { asFragment } = renderComponent({
    template: `
    <Breadcrumb separator="-">
      <BreadcrumbItem>
        <BreadcrumbLink href="/">Home</BreadcrumbLink>
      </BreadcrumbItem>
      <BreadcrumbItem>
        <BreadcrumbLink href="/about">About</BreadcrumbLink>
      </BreadcrumbItem>
      <BreadcrumbItem isCurrentPage>
        <BreadcrumbLink href="/contact">Contact</BreadcrumbLink>
      </BreadcrumbItem>
    </Breadcrumb>`
  })
  expect(asFragment()).toMatchSnapshot()
})
