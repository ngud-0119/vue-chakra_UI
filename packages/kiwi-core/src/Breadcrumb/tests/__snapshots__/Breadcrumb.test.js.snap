// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display custom seperator  1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="css-0"
  >
    <ol
      class="css-0"
    >
      <li
        class="css-18biwo"
      >
        <a
          class="css-1kvwaf7 css-7ey7in"
          href="/"
        >
          Home
        </a>
        <span
          class="css-dp0f2t"
          role="presentation"
        >
          -
        </span>
      </li>
      <li
        class="css-18biwo"
      >
        <a
          class="css-1kvwaf7 css-7ey7in"
          href="/about"
        >
          About
        </a>
        <span
          class="css-dp0f2t"
          role="presentation"
        >
          -
        </span>
      </li>
      <li
        class="css-18biwo"
      >
        <span
          aria-current="page"
          class="css-0"
          href="/contact"
        >
          Contact
        </span>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div>
    <nav
      aria-label="breadcrumb"
      class="css-0"
    >
      <ol
        class="css-0"
      >
        <li
          class="css-18biwo"
        >
          <a
            class="css-1kvwaf7 css-7ey7in"
            href="#"
          >
            Breadcrumb 1
          </a>
          <span
            class="css-6eovvf"
            role="presentation"
          >
            /
          </span>
        </li>
        <li
          class="css-18biwo"
        >
          <a
            class="css-1kvwaf7 css-7ey7in"
            href="#"
          >
            Breadcrumb 2
          </a>
          <span
            class="css-1hgjpgb"
            role="presentation"
          >
            /
          </span>
        </li>
      </ol>
    </nav>
  </div>
</DocumentFragment>
`;
